packages/blade/src/components/Drawer/_decisions/decisions.md

```diff
@@ -0,0 +1,147 @@
+# Drawer
+
+A drawer is a panel that slides in mostly from right side of the screen over the existing content in the viewport. It helps in providing additional details or context and can also be used to promote product features or new products.
+
+<img width="500px" src="./2024-02-07-20-04-07.png" alt="Drawer Figma Skeleton">
+
+## Design
+
+- [Drawer - Figma Design](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade-DSL?node-id=78667%3A66663&mode=dev)
+
+## Proposed API
+
+```jsx
+<Drawer
+  isOpen={}
+  onDismiss={() => {}}
+  showOverlay
+>
+  <DrawerHeader
+    title=""
+    subtitle=""
+    leading={<DrawerHeaderIcon />}
+    titleSuffix={<DrawerHeaderBadge />}
+    trailing={
+      <>
+        <DrawerHeaderLink />
+        <DrawerHeaderButton />
+      </>
+    }
+  />
+  <DrawerBody>
+    <Slot />
+  </DrawerBody>
+<Drawer>
+```
+
+<details>
+<summary><b>Full Usage Example</b></summary>
+
+```jsx
+const MyCuteDrawer = () => {
+  const [showDrawer, setShowDrawer] = React.useState(false);
+  return (
+    <Box>
+      <Button onClick={() => setShowDrawer(true)}>Open Drawer</Button>
+      <Drawer
+        isOpen={showDrawer}
+        onDismiss={() => {
+          setShowDrawer(false);
+        }}
+      >
+        <DrawerHeader
+          title="Announcements"
+        />
+        <DrawerBody>
+          <FTXAnnouncement />
+          <RazorpayOnePromotions />
+          <CatPictures />
+        </DrawerBody>
+      <Drawer>
+    </Box>
+  )
+}
+
+```
+
+</details>
+
+### Props
+
+```ts
+type DrawerProps = {
+  /**
+   * Controlled state of drawer open or not
+   */
+  isOpen: boolean;
+
+  /**
+   * Dismiss handler
+   */
+  onDismiss: () => void;
+
+  /**
+   * Show or hide overlay.
+   *
+   * Also decides if clicking outside on overlay closes the drawer or not
+   */
+  showOverlay?: boolean;
```

Comment:
So overlay is optional? Won't that look weird? What is the use case?

Reply:
We have usecases where Drawer is used for Announcements. In those cases, users should still be able to interact with dashboard items even with Drawer open so we cant show overlay or hide drawer on clicking outside.

<img width="679" alt="image" src="https://github.com/razorpay/blade/assets/30949385/58dbc3c6-d12f-42cf-bb6a-83f321b3d0f9">

Reply:
when overlay is disabled, does the area outside the drawer remain interactive?

Reply:
yes

Reply:
> does the area outside the drawer remain interactive

We should ideally control this with a new prop like `shouldDismissOnClickOutside`